sdk/pinocchio/src/account_info.rs

```diff
@@ -172,6 +192,15 @@ impl AccountInfo {
         core::ptr::write_volatile(&(*self.raw).owner as *const _ as *mut Pubkey, *new_owner);
     }
 
+    /// Return true if the account borrow state is set to the given state.
+    ///
+    /// This will test both data and lamports borrow state.
+    #[inline(always)]
+    pub fn check_borrowed_state(&self, state: BorrowState) -> bool {
```

Comment:
nit: up to you, but `is_borrowed_with_state` or `is_borrowed_for_state` or just `is_borrowed` might be a clearer name

Reply:
Good call, `is_borrowed` sounds good. Now it makes more sense that the method returns a `bool` – the "check" kind of implied that it would return a `Result`.